code/game/objects/items/weapons/armblade.dm

```diff
@@ -15,9 +15,104 @@
 	attack_verb = list("attacks", "slashes", "rends", "slices", "tears", "rips", "shreds", "cuts")
 	hitsound = "sound/weapons/bloodyslice.ogg"
 	cant_drop = 1
+	var/spin_last_used
+	var/spin_cooldown = 30 SECONDS
+	//Spin attack variables
+	var/spin_duration = 1.5 SECONDS //How many seconds the spin lasts
+	var/attack_modulo = 0.5 SECONDS //How often everyone nearby is attacked, every X seconds
+	var/step_modulo = 0.2 SECONDS //How often the changeling moves one step, every X seconds
+	var/sound_modulo = 0.8 SECONDS //How often the sound effect plays, every X seconds, current audio file lasts ~0.8 seconds
+
+
+/obj/item/weapon/armblade/New()
+	..()
+	spin_last_used = world.timeofday //Always the latest on New() so that one can't just remake the armblade.
 
 /obj/item/weapon/armblade/IsShield()
     return 1
 
 /obj/item/weapon/armblade/dropped()
 	qdel(src)
+
+/obj/item/weapon/armblade/examine(mob/user, size, show_name)
+	..()
+	if(ischangeling(user) && is_holder_of(user, src))
+		if((spin_last_used + spin_cooldown) <= world.timeofday)
+			to_chat(user, "<span class='good'>The spin attack is ready to be used! Activate it in your hand to initiate it.</span>")
+		else
+			to_chat(user, "<span class='warning'>The spin attack is not available yet! It will be available in [((spin_last_used + spin_cooldown) - world.timeofday)/10] seconds.</span>")
+
+/obj/item/weapon/armblade/attack_self(mob/user)
+	..()
+	if(!ischangeling(user))
+		return
+	if((spin_last_used + spin_cooldown) <= world.timeofday)
+		if(user.incapacitated()) //Sanity
+			to_chat(user, "<span class='warning'>You cannot move your armblade around!</span>")
+			return
+		if(user.locked_to) //Can't use while buckled up AKA no chairs, vehicles or anything like that
+			to_chat(user, "<span class='warning'>You need to be on foot to perform the spin attack!</span>")
+			return
+		if(!isturf(user.loc)) //In a mecha, locker or something else
+			to_chat(user, "<span class='warning'>You do not have enough room to move your armblade around!</span>")
+			return
+		spin_attack(user)
+	else
+		to_chat(user, "<span class='warning'>Your arm is too exhausted to perform the spin attack! It will be available in [((spin_last_used + spin_cooldown) - world.timeofday)/10] seconds.</span>")
+
+/obj/item/weapon/armblade/proc/spin_attack(var/mob/user)
+	var/initial_direction = user.dir //Direction in which the changeling will move
+	var/spin_direction = (user.active_hand == GRASP_RIGHT_HAND) ? "Left" : "Right" //Different spinning directions depending on arm
+	var/spin_facing = initial_direction //For the purpose of where to spin next
+	var/delay_track = spin_duration //If the action ends prematurely for some reason it will free the changeling of the remaining duration
+
+	visible_message("<span class='sinister'>[user] starts wildly spinning their armblade around!</span>")
+	user.delayNextMove(spin_duration, 1) //Can't move during the spin
```

Comment:
Rather than this, I would just make it register an event that cancels the spin if you move, so that you're not locking yourself out of control for 1.5 seconds (+/- lag)